back/engines/free/user_confirmation/app/interactors/user_confirmation/reset_user_email.rb

```diff
@@ -0,0 +1,17 @@
+module UserConfirmation
+  class ResetUserEmail < ApplicationInteractor
+    delegate :user, to: :context
+    delegate :new_email, to: :context, allow_nil: true
+
+    def call
+      return unless new_email
+
+      context.old_email = user.email
+      fail_with_error!(user.errors) unless user.reset_email(new_email)
+    end
+
+    def rollback
+      user.update(email: context.old_email)
```

Comment:
Would it make sense to do `update!` instead of `update` to avoid silent failures when trying to restore the old email? (if for some reason this would fail, I think we need to be notified of that e.g. trough Sentry)

Reply:
Good point 👍